### PR TITLE
Ad budget refactor

### PR DESCRIPTION
**Pull request by apalski**
PR to remove budget_type from budget model
- Remove budget_type from budget table, model and controller params
- Remove uniqueness of name from budget model
- Remove validation for uniqueness of name from model spec
- Remove budget_type default specs from model spec
- Remove index view
- Remove expense partial
- Refactor show view for removing expense partial
- Refactor controller and feature specs to remove budget_type references
- Refactor delete feature spec to allow for current_user ownership
All feedback and improvement suggestions welcome!
https://trello.com/c/Q5xdEuA8
https://trello.com/c/01mPrV21
Thanks for taking a look!

### CODE REVIEW

**Comment by apalski** [comment]
Merging now, further feedback can be added to this PR and will be reviewed thank you!

File: app/models/budget.rb [hunk 1]
```diff
@@ -3,5 +3,9 @@ class Budget < ActiveRecord::Base
   has_many :incomes
   has_many :expenses
 
-  validates :name, presence: true, uniqueness: true
+  enum frequency: { weekly: 0, fortnightly: 1, monthly: 2 }
+
+  validates :name, presence: true
+  validates :frequency, presence: true
+  validates :frequency, inclusion: { in: Budget.frequencies.keys }
```
Comment (elle) [inline app/models/budget.rb]: Lines 9 and 10 can be merged to one line

File: config/locales/en.yml [hunk 2]
```diff
@@ -47,9 +47,15 @@ en:
...
+    attributes:
+      frequencies:
+        weekly: "weekly"
+        fortnightly: "fortnightly"
+        monthly: "monthly"
     defaults:
+      frequency: "Frequency"
       name: "Name"
-      budget_type: "Budget Type"
+      require_budget: "You don't have a budget"
```
Comment (elle) [inline config/locales/en.yml]: Maybe change the message to be more informative: "you need to have a budget to be able to use this app"

File: spec/controllers/budgets_controller_spec.rb [hunk 3]
```diff
@@ -80,8 +104,8 @@
 
       it "renders edit" do
         user = create(:user)
-        budget = create(:budget, user: user)
         allow(controller).to receive(:current_user).and_return(user)
+        budget = create(:budget, user: user)
```
Comment (elle) [inline spec/controllers/budgets_controller_spec.rb]: I usually create the objects and then stub the controller but really up to you

**Comment by elle** [comment]
Would be good to include in the PR description why things were done as well. The reason behind the changes

File: app/models/budget.rb (hunk 1 above)
Comment (apalski) [reply app/models/budget.rb]: Thanks Elle, done in another PR, ad-budget-frequency-amounts, sha: 5ed2ca8

File: config/locales/en.yml (hunk 2 above)
Comment (apalski) [reply config/locales/en.yml]: Done in PR, ad-css-styling, sha: 0723be2

File: spec/controllers/budgets_controller_spec.rb (hunk 3 above)
Comment (apalski) [reply spec/controllers/budgets_controller_spec.rb]: No, I'd rather do it the correct, more sensible way, thanks Elle. Done in PR ad-budget-frequency-amounts sha: 292cc37

**Comment by apalski** [comment]
That makes sense Elle, I will try to incorporate reasons into future PRs

**Pull request merged**